server/app/services/applicant/ApplicantService.java

```diff
@@ -835,13 +835,31 @@ public CompletionStage<ApplicationPrograms> relevantProgramsForApplicant(long ap
   }
 
   /**
-   * Returns whether an application is maybe eligible for a program, and empty if there are no
-   * eligibility conditions for the program.
+   * Returns whether an applicant is maybe eligible for a program based on their latest answers, and
```

Comment:
**nitpick:** This is not necessarily whether an `applicant` may be eligible because the applicantData of the application could be passed through as well. You could have this function accept an "applicant" instead of applicantData and then do getApplicantData on the applicant to ensure this is working as expected? Thoughts?

Reply:
Good point! I think that's a good idea; one question though -- this is called in `relevantProgramsForApplicant()` with `draftApp.getApplicant().getApplicantData()`. Any thoughts on whether or not for draft applications we should use the application.applicantData or the applicant.applicantData?

Reply:
I _think_ we should use the draftApp.getApplicant().getApplicantData(). That's because if the admin gates on eligibility < 30 and I go in as an applicant fill out `application 1` that my age is 29 then I abandon it because I'm not eligible and I go to fill out `application 2` a few months later and fill in that my age is 30, I'd like to go to the home page and know that I'm now eligible for `application 1` too, but if we look at the application.getApplicantData the application that hasn't been submitted won't be looking at my latest answer for eligibility. But I think if I opened that application it would sync with what I answered most recently (in application 2). So basically for unsubmitted applications in the applicant UI, I think it should look at the most recent applicant data, but on the admin side it should be looking at the application data. Does that make sense?